libraries/botbuilder/src/fileTranscriptStore.ts

```diff
@@ -256,28 +255,29 @@ export class FileTranscriptStore implements TranscriptStore {
     /**
      * @private
      */
-    private getActivityFilename(activity: Activity): string {
-        return `${getTicks(activity.timestamp)}-${this.sanitizeKey(activity.id)}.json`;
+    private async getActivityFilename(activity: Activity): Promise<string> {
+        return `${getTicks(activity.timestamp)}-${await this.sanitizeKey(activity.id)}.json`;
     }
 
     /**
      * @private
      */
-    private getChannelFolder(channelId: string): string {
-        return join(this.rootFolder, this.sanitizeKey(channelId));
+    private async getChannelFolder(channelId: string): Promise<string> {
+        return join(this.rootFolder, await this.sanitizeKey(channelId));
     }
 
     /**
      * @private
      */
-    private getTranscriptFolder(channelId: string, conversationId: string): string {
-        return join(this.rootFolder, this.sanitizeKey(channelId), this.sanitizeKey(conversationId));
+    private async getTranscriptFolder(channelId: string, conversationId: string): Promise<string> {
+        return join(this.rootFolder, await this.sanitizeKey(channelId), await this.sanitizeKey(conversationId));
     }
 
     /**
      * @private
      */
-    private sanitizeKey(key: string): string {
-        return filenamify(key);
+    private async sanitizeKey(key: string): Promise<string> {
+        const filenamify = await import('filenamify');
```

Comment:
Maybe this could work https://github.com/nktnet1/import-sync, and use it at the top level file like the rest of the imports, and we avoid using async, it would be great.
And maybe we dont need tsup